app/code/Magento/StoreGraphQl/Model/Resolver/Store/StoreConfigDataProvider.php

```diff
@@ -55,24 +74,42 @@ public function __construct(
      */
     public function getStoreConfigData(StoreInterface $store): array
     {
-        $storeConfigData = array_merge(
-            $this->getBaseConfigData($store),
-            $this->getExtendedConfigData((int)$store->getId())
-        );
-        return $storeConfigData;
+        $defaultStoreConfig = $this->storeConfigManager->getStoreConfig($store);
+        return $this->prepareStoreConfigData($defaultStoreConfig);
     }
 
     /**
-     * Get base config data
+     * Get website available stores
      *
      * @param StoreInterface $store
      * @return array
      */
-    private function getBaseConfigData(StoreInterface $store) : array
+    public function getAvailableStoreConfig(StoreInterface $store): array
     {
-        $storeConfig = current($this->storeConfigManager->getStoreConfigs([$store->getCode()]));
+        $storeIds = $this->storeWebsiteRelation->getStoreByWebsiteId($store->getWebsiteId());
```

Comment:
Could we add a method to \Magento\Store\Model\ResourceModel\StoreWebsiteRelation that return the code instead of id (e.g. getStoreCodesByWebsiteId), to get the store codes.
then use StoreConfigManager::getStoreConfigs($code) to get the config info.
then we can avoid changing the protected method